projects/packages/sync/src/modules/class-woocommerce-hpos-orders.php

```diff
@@ -303,22 +303,48 @@ private function filter_order_data( $order_object ) {
 		if ( '' === $filtered_order_data['status'] ) {
 			$filtered_order_data['status'] = 'pending';
 		}
+		$filtered_order_data['status'] = $this->get_wc_order_status_with_prefix( $filtered_order_data['status'] );
 
 		return $filtered_order_data;
 	}
 
 	/**
-	 * Returns all possible order status keys using 'wc_get_order_statuses', if possible..
+	 * Returns all possible order status keys, including 'auto-draft' and 'trash'.
 	 *
 	 * @access protected
 	 *
-	 * @return array Filtered order metadata.
+	 * @return array An array of all possible status keys, including 'auto-draft' and 'trash'.
 	 */
 	protected function get_all_possible_order_status_keys() {
-		$order_statuses    = array( 'checkout-draft', 'auto-draft', 'trash' );
+		$order_statuses    = array( 'auto-draft', 'trash' );
```

Comment:
@vedanshujain Just a confirmation if we should include `draft` here. In our discussion you mentioned that `draft` is a potential order status, however based on the [Woo check upon setting the status](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/abstracts/abstract-wc-order.php#L631), it looks like it's not.

Reply:
yeah, looks like we are actively changing draft to pending in case it is set. so this implementation looks good,